src/Widgets/PopoverWidget.vala

```diff
@@ -92,16 +92,14 @@ public class BluetoothIndicator.Widgets.PopoverWidget : Gtk.Box {
 
 
         object_manager.device_added.connect ((device) => {
+            // Remove existing rows for this device
+            remove_device (device);
```

Comment:
Seems kinda of silly to remove the device and then add it again. If the device row already exists, we should probably just return and do nothing

Reply:
But the old one isn't connected anymore. It at least didn't display the right state anymore when I tried such an approach. (It's also weird that the one that isn't connected anymore isn't removed)

Reply:
Fair enough! Can you leave a comment so we know that this is done this way for a reason and someone doesn’t try to “clean up” later?

Reply:
Sure thing. I hope this is clear enough.